rotkehlchen/api/rest.py

```diff
@@ -4675,15 +4678,35 @@ def export_history_events(
             )
 
         if len(history_events) == 0:
-            return api_response(
-                wrap_in_fail_result('No history processed in order to perform an export'),
+            return wrap_in_fail_result(
+                message='No history processed in order to perform an export',
                 status_code=HTTPStatus.CONFLICT,
             )
 
+        with self.rotkehlchen.data.db.conn.read_ctx() as cursor:
+            settings = self.rotkehlchen.get_settings(cursor)
+            currency = settings.main_currency.resolve_to_asset_with_oracles()
+
         serialized_history_events = []
         headers: dict[str, None] = {}
         for event in history_events:
-            serialized_event = event.serialize_for_csv()
+            if event.balance.usd_value != ZERO and currency != A_USD:
+                try:
+                    price = self.get_rate_in_profit_currency(
+                        asset=event.asset,
+                        timestamp=ts_ms_to_sec(event.timestamp),
+                        currency=currency,
+                    )
+                except (PriceQueryUnsupportedAsset, RemoteError):
+                    price = ZERO_PRICE
+                except NoPriceForGivenTimestamp as e:
+                    # In the case of NoPriceForGivenTimestamp when we got rate limited we let
+                    # it propagate so the user can take action after the report is made
+                    if e.rate_limited is True:
+                        raise
+                    price = ZERO_PRICE
+                event.balance.usd_value = event.balance.amount * price
```

Comment:
this looks wrong. here is not the usd price but the price in the main currency. THis is unexpected and can lead to errors